src/components/Nfts/hooks/useFetchNfts.tsx

```diff
@@ -0,0 +1,32 @@
+import { useMemo } from 'react'
+import { useGetNftUserTokensQuery } from 'state/apis/nft/nftApi'
+import { selectWalletAccountIds } from 'state/slices/common-selectors'
+import { useAppSelector } from 'state/store'
+
+export const useFetchNfts = () => {
+  const requestedAccountIds = useAppSelector(selectWalletAccountIds)
+
+  const {
+    isUninitialized,
+    isLoading: isNftUserTokensQueryLoading,
+    isFetching,
+    data,
+  } = useGetNftUserTokensQuery(
+    {
+      accountIds: requestedAccountIds,
+    },
+    { skip: requestedAccountIds.length === 0 },
+  )
+
+  return useMemo(
+    () => ({
+      isLoading:
+        requestedAccountIds.length === 0 ||
+        isUninitialized ||
```

Comment:
Is considering query not-yet-fired as loading intended here?

Reply:
correct yeah, since this is a hydration hook we want the app to treat it as loading until a result arrives